storage/cache/subtree_cache.go

```diff
@@ -326,15 +326,23 @@ func (s *SubtreeCache) SetNodeHash(id storage.NodeID, h []byte, getSubtree GetSu
 	if c.Prefix == nil {
 		return fmt.Errorf("nil prefix for %v (key %v)", id.String(), prefixKey)
 	}
-	s.dirtyPrefixes[prefixKey] = true
 	// Determine whether we're being asked to store a leaf node, or an internal
 	// node, and store it accordingly.
 	sfxKey := sx.String()
 	if int32(sx.Bits) == c.Depth {
+		// Short-circuit write if we're simply overwriting an identical value.
```

Comment:
Suggest rewording this for morer clarity. How about "Cache state is unchanged if we're overwriting an identical value" or summat?

Reply:
That's a bit too opaque in my view, that makes it sound a bit like the only change is in the cache (and there's a silent implication, _if_ you know the code well enough, that no writes will be performed).

Reply:
OK I just thought "short circuit" could be ambiguous.